acme/db/nosql/account.go

```diff
@@ -259,21 +259,42 @@ func (db *DB) DeleteExternalAccountKey(ctx context.Context, provisionerName, key
 }
 
 // GetExternalAccountKeys retrieves all External Account Binding keys for a provisioner
-func (db *DB) GetExternalAccountKeys(ctx context.Context, provisionerName string) ([]*acme.ExternalAccountKey, error) {
+func (db *DB) GetExternalAccountKeys(ctx context.Context, provisionerName, cursor string, limit int) ([]*acme.ExternalAccountKey, string, error) {
 	entries, err := db.db.List(externalAccountKeyTable)
 	if err != nil {
-		return nil, err
+		return nil, "", err
+	}
+
+	// set sane limits; based on the Admin API limits
+	switch {
+	case limit <= 0:
+		limit = 20
+	case limit > 100:
+		limit = 100
 	}
 
+	foundCursorKey := false
 	keys := []*acme.ExternalAccountKey{}
-	for _, entry := range entries {
+	for _, entry := range entries { // entries is sorted alphabetically on the key (ID) of the EAK; no need to sort this again.
 		dbeak := new(dbExternalAccountKey)
 		if err = json.Unmarshal(entry.Value, dbeak); err != nil {
-			return nil, errors.Wrapf(err, "error unmarshaling external account key %s into ExternalAccountKey", string(entry.Key))
+			return nil, "", errors.Wrapf(err, "error unmarshaling external account key %s into ExternalAccountKey", string(entry.Key))
 		}
 		if dbeak.Provisioner != provisionerName {
 			continue
 		}
+		// skip the IDs not matching the cursor to look for in the sorted list.
```

Comment:
Are we confident that `list` returns a consistent order of items?

Reply:
`bbolt` uses byte-sorted ordering (on key), `badger` is sorted lexicographically (on key) and the `mysql` implementation (most likely) returns the results in order of primary key ascending, if I'm correct. Within a single call to `List`, and thus within a single database transaction, the ordering of the items is consistent. 

It's a different story if we expect multiple systems calling the APIs for creation and retrieval of EAKs at the same time. The keys are random, so new items will get a spot in between other items between calls to List. While paging through the items, this may thus cause an inconsistent view of the data. If we want to prevent this, we'd need to add an index using `CreatedAt` as the key and loop through the items based on that ordering. WDYT?